source/common/runtime/runtime_impl.cc

```diff
@@ -232,17 +232,19 @@ void SnapshotImpl::walkDirectory(const std::string& path, const std::string& pre
 
       // Read the file and remove any comments. A comment is a line starting with a '#' character.
       // Comments are useful for placeholder files with no value.
-      const std::vector<std::string> lines =
-          StringUtil::split(Filesystem::fileReadToEnd(full_path), "\n");
-      for (const std::string& line : lines) {
-        if (!line.empty() && line.at(0) == '#') {
+      auto text_file{Filesystem::fileReadToEnd(full_path)};
+      auto lines = StringUtil::splitToken(text_file, "\n");
+      for (auto line : lines) {
+        if (!line.empty() && line.front() == '#') {
           continue;
         }
-        entry.string_value_ += line + "\n";
+        if (line == lines.back()) {
+          entry.string_value_.append(std::string(StringUtil::rtrim(line)));
```

Comment:
Here I think it would be faster to append the bytes directly, rather than make a temporary string and then append that.
  absl::string_view trimmed = StringUtil::rtrim(line);
  entry.string_value_.append(trimmed.data(), trimmed.size());

Reply:
💯